lib/measured/cache/json.rb

```diff
@@ -0,0 +1,48 @@
+module Measured::Cache
+  class Json
+    attr_reader :filename, :path
+
+    def initialize(filename)
+      @filename = filename
+      @path = Pathname.new(File.join(File.dirname(__FILE__), "../../../cache", @filename)).cleanpath
+    end
+
+    def exist?
+      File.exist?(@path)
+    end
+
+    def read
+      return nil unless exist?
+      decode(JSON.load(File.read(@path)))
+    end
+
+    def write(table)
+      raise ArgumentError, "Cannot overwrite file cache at runtime."
+    end
+
+    private
+
+    # JSON dump and load of Rational objects exists, but it changes the behaviour of JSON globally if required.
+    # Instead, the same marhalling technique is rewritten here to prevent changing this behaviour project wide.
+    # https://github.com/ruby/ruby/blob/trunk/ext/json/lib/json/add/rational.rb
```

Comment:
Compromise.

JSON serialization of a rational returns a string: 
```ruby
Rational(2, 3)
> "2/3"
```

Then re-parsing it gets back the string, not the number.

There exists a typed serializer for Rational in the JSON core lib, but it changes the behaviour globally, which is unacceptable. Instead I write our own little serializer and deserializer that stores the rational as a hash of numerator and denominator.